aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -18,41 +18,51 @@ module AptosFramework::Coin {
     const ENO_MINT_CAPABILITY: u64 = 7;
     const EDESTRUCTION_OF_NONZERO_TOKEN: u64 = 8;
 
-    // Core data structures
+    /// Core data structures
 
-    // Represents a set amount of coin
+    /// Main structure representing a coin/token in an account's custody.
     struct Coin<phantom CoinType> has store {
+        /// Amount of coin this address has.
         value: u64,
     }
 
-    // Represents ownership of coin
+    /// A holder of a specific coin type. This is a wrapper that allows
+    /// adding more data such as events and can be thought of as a "wallet".
     struct CoinStore<phantom CoinType> has key {
         coin: Coin<CoinType>,
         deposit_events: EventHandle<DepositEvent>,
         withdraw_events: EventHandle<WithdrawEvent>,
     }
 
-    // Represents resources associated with a coin
+    /// Information about a specific coin type declared on the coin creator.
     struct CoinInfo<phantom CoinType> has key {
         name: ASCII::String,
-        scaling_factor: u64,
+        /// Symbol of the coin, usually a shorter version of the name.
+        symbol: ASCII::String,
+        /// Number of decimals used to get its user representation/
```

Comment:
```suggestion
        /// Number of decimals used to get its user representation/
```
The doc you sent me for erc20 was a lot more exhaustive here. I would probably rewrite this to emphasize that this is the left shifted amount that represents a single unit. For example, a 6 here would translate 1 real world unit to 1,000,000 on-chain.

Reply:
Updated